sources/api/migration/migrations/v1.11.0/credential-providers/src/main.rs

```diff
@@ -0,0 +1,24 @@
+#![deny(rust_2018_idioms)]
+
+use migration_helpers::common_migrations::AddSettingsMigration;
+use migration_helpers::{migrate, Result};
+use std::process;
+
+/// We added a new setting for configuring kubelet's image credential
+/// provider plugins. Initially this is only to support ecr-credential-provider,
+/// but others may be added as needed.
+fn run() -> Result<()> {
+    migrate(AddSettingsMigration(&[
+        "settings.kubernetes.credential-providers",
+    ]))
+}
```

Comment:
This should be an `AddPrefixesMigration` since there will be a cluster of settings underneath.